mantle/cmd/kola/testiso.go

```diff
@@ -401,116 +446,64 @@ func runTestIso(cmd *cobra.Command, args []string) error {
-		duration, err := testLiveIso(ctx, instIso, filepath.Join(outputDir, scenarioISOInstall), false, false)
-		printResult(scenarioISOInstall, duration, err)
-		if err != nil {
-			return err
-		}
-	}
-	if _, ok := targetScenarios[scenarioISOOfflineInstall]; ok {
-		if kola.CosaBuild.Meta.BuildArtifacts.LiveIso == nil {
-			return fmt.Errorf("build %s has no live ISO", kola.CosaBuild.Meta.Name)
+		fmt.Printf("Running test: %s\n", test)
+		components := strings.Split(test, ".")
+
+		if kola.HasString("4k", components) {
+			enable4k = true
 		}
-		ranTest = true
-		instIso := baseInst // Pretend this is Rust and I wrote .copy()
-		duration, err := testLiveIso(ctx, instIso, filepath.Join(outputDir, scenarioISOOfflineInstall), true, false)
-		printResult(scenarioISOOfflineInstall, duration, err)
-		if err != nil {
-			return err
+		if kola.HasString("nm", components) {
+			addNmKeyfile = true
 		}
-	}
-	if _, ok := targetScenarios[scenarioISOLiveLogin]; ok {
-		if kola.CosaBuild.Meta.BuildArtifacts.LiveIso == nil {
-			return fmt.Errorf("build %s has no live ISO", kola.CosaBuild.Meta.Name)
+		if kola.HasString("mpath", components) {
+			enableMultipath = true
 		}
-		ranTest = true
-		duration, err := testLiveLogin(ctx, filepath.Join(outputDir, scenarioISOLiveLogin))
-		printResult(scenarioISOLiveLogin, duration, err)
-		if err != nil {
-			return err
+		if kola.HasString("uefi-secure", components) {
+			enableUefiSecure = true
+		} else if kola.HasString("uefi", components) {
+			enableUefi = true
 		}
-	}
-	if _, ok := targetScenarios[scenarioISOAsDisk]; ok {
-		if kola.CosaBuild.Meta.BuildArtifacts.LiveIso == nil {
-			return fmt.Errorf("build %s has no live ISO", kola.CosaBuild.Meta.Name)
+		if kola.HasString("offline", components) {
+			isOffline = true
+		} else if kola.HasString("online", components) {
+			isOffline = false
 		}
-		switch coreosarch.CurrentRpmArch() {
-		case "x86_64":
-			ranTest = true
-			duration, err := testAsDisk(ctx, filepath.Join(outputDir, scenarioISOAsDisk))
-			printResult(scenarioISOAsDisk, duration, err)
-			if err != nil {
-				return err
-			}
+
+		switch components[0] {
+		case "pxe-offline-install", "pxe-online-install":
+			duration, err = testPXE(ctx, inst, filepath.Join(outputDir, test))
+		case "iso-as-disk":
+			duration, err = testAsDisk(ctx, filepath.Join(outputDir, test))
+		case "iso-live-login":
+			duration, err = testLiveLogin(ctx, filepath.Join(outputDir, test))
+		case "iso-install":
+			duration, err = testLiveIso(ctx, inst, filepath.Join(outputDir, test), false)
+		case "iso-offline-install":
+			duration, err = testLiveIso(ctx, inst, filepath.Join(outputDir, test), false)
+		case "miniso-install":
+			duration, err = testLiveIso(ctx, inst, filepath.Join(outputDir, test), true)
 		default:
-			// no hybrid partition table to boot from
-			fmt.Printf("%s unsupported on %s; skipping\n", scenarioISOAsDisk, coreosarch.CurrentRpmArch())
+			plog.Fatal("Unknown test name:", test)
```

Comment:
do we need the `%s` here? If not that's cool just wanted to check.

```suggestion
			plog.Fatal("Unknown test name: %s", test)
```

Reply:
Go was complaining about it:
`cmd/kola/testiso.go:504:4: (*github.com/coreos/pkg/capnslog.PackageLogger).Fatal call has possible formatting directive %s`

 `plog.Fatal` seems to not allow `%s`.

Reply:
Needs to be `plot.Fatalf`